pkg/policy/l4.go

```
@@ -964,10 +964,12 @@ func createL4Filter(policyCtx PolicyContext, peerEndpoints api.EndpointSelectorS
 		Ingress:             ingress,
 	}
 
-	if peerEndpoints.SelectsAllEndpoints() {
+	es := api.EndpointSelectorSlice(types.FromEndpointSelectorInterfaceSlice[api.EndpointSelector](peerEndpoints))
+	if es.SelectsAllEndpoints() {
 		l4.wildcard = l4.cacheIdentitySelector(api.WildcardEndpointSelector, origin.stringLabels(), selectorCache)
 	} else {
-		l4.cacheIdentitySelectors(peerEndpoints, origin, selectorCache)
+		l4.cacheIdentitySelectors(es, origin, selectorCache)
+		fqdns := types.FromEndpointSelectorInterfaceSlice[api.FQDNSelector](peerEndpoints)
 		l4.cacheFQDNSelectors(fqdns, origin, selectorCache)
 	}
```

This seems to iterate and handle all endpoint selectors in `peerEndpoints` that have types `api.EndpointSelector` and `api.FQDNSelector`. Grepping for `FromEndpointSelectorInterfaceSlice`, I also see that function being used for `api.CIDRRule` and `api.CIDR`. Does this mean that creating an L4 filter could omit CIDR-based peer selectors? Does this logic need to be extended for other types as well? If yes, how do we avoid silently dropping valid endpoint selector types with this new generic structure?

---

Short answer from [the other thread](https://github.com/cilium/cilium/pull/40213#discussion_r2373563729) is that the CIDR-based selectors are being converted into `api.EndpointSelector` as they're put into `peerEndpoints` so the `es` unboxing here should include CIDR selectors as well.

(If the PR broke CIDR selection, I would think that at least some of the tests should have failed so I don't think there's much chance of regression here).